debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -160,6 +160,14 @@ else if (actualColumn.jdbcType() == Types.DATE) {
         else if (actualColumn.jdbcType() == Types.TIMESTAMP) {
             return readTimestampField(rs, fieldNo, actualColumn, actualTable);
         }
+        // JDBC's rs.GetObject() will return a Boolean for all TINYINT(1) columns.
+        // TINYINT columns are reprtoed as SMALLINT by JDBC driver
+        else if (actualColumn.jdbcType() == Types.TINYINT || actualColumn.jdbcType() == Types.SMALLINT) {
```

Comment:
Ok, read up a bit more on this. Seems we could achieve the same without any real code change [by setting](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html) `tinyInt1isBit` to false (it's true by default). Both should work, leaving it to you what you prefer. In any case it should be a workaround for the issue reporter until the fix has been shipped.